PlatformMessage platform
PathMessage activePath
PathMessage[] paths